### DISCUSSION TITLE
When you kill uvicorn, the child processes stick around as zombies.

### DISCUSSION THREAD

**Discussion by gnat** [Kludex/uvicorn#2369] [Potential Issue]
Started happening after https://github.com/encode/uvicorn/pull/2183

**Comment by Kludex** [comment]
I questioned this at the time. It looks like the wanted behavior?

**Comment by gnat** [comment]
In what scenario would it be beneficial behavior?


* The PID's are lost, the only reasonable way to stop the child zombies is to kill all python processes (very bad).
* No way to recover the workers with another uvicorn.

**Reply by Kludex** [reply]
@abersheeran

**Comment by abersheeran** [comment]
Did you use the kill -9 signal?

**Reply by abersheeran** [reply]
If you can provide a way to reproduce it, that would speed up my resolution of this issue.

**Reply by abersheeran** [reply]
https://github.com/encode/uvicorn/blob/d79f285184404694c77f7ca649858e7488270cf7/uvicorn/supervisors/multiprocess.py#L156

Because it seems that killing the uvicorn master with a regular signal does not produce zombie processes.

**Comment by abersheeran** [comment]
https://github.com/encode/uvicorn/pull/2371 Could you try this PR? I'm not sure if you have the same problem as mine.
